app/models/piece.rb

```diff
@@ -48,4 +48,32 @@ def valid_move?(x_destination, y_destination)
   def diff(destination, current)
     (destination - current).abs
   end
+
+
+  # Capture logic
+  # returns true if the move is valid
+  def move_to!(x_destination, y_destination)
+    return true if valid_move?(x_destination, y_destination)
+    capture_piece(x_destination, y_destinaiton)
+    game.update_attributes(x_position: x, y_position: y)
+  end
+
+  # returns false if square is occupied
+  # returns true if piece is captured
+  def capture_piece(x_destination, y_destination)
+    return false if game.occupied?(x_destination, y_destination)
+    self.game.find_piece(x_destination, y_destination).captured!
+    return true
+  end
+
+  def occupied?(x_destination, y_destination)
+    pieces.where(game_id: game.id, y_destination: y, x_destination: x).any?
```

Comment:
This this will have to be more like `Piece.where(game_id: game.id, y_position: y_destination, x_position: x_destination).any?`